tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -0,0 +1,324 @@
+# -*- coding: utf-8 -*-
```

Comment:
@arjundc-db Can we add unit tests for `log_artifact()/`log_artifacts()`, `download_artifacts()`, and `list_artifacts()` that construct a `DatabricksArtifactRepo` rooted at a subdirectory of the run artifact root (e.g., `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts/my/path`) and ensure that these APIs operate on paths relative to this subdirectory? (E.g., `list_artifacts('foo')` should list artifacts at `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts/my/path/foo` rather than `dbfs:/databricks/mlflow-tracking/<EXP_ID>/<RUN_ID>/artifacts/foo`)

To test this, we should be able to mock out `_upload_to_cloud`, `_call_endpoint`, and `_download_from_cloud` and ensure that they're called with the expected paths / request bodies. Let me know if this makes sense!